homeassistant/components/switch/rpi_rf.py

```diff
@@ -101,22 +105,23 @@ def is_on(self):
         """Return true if device is on."""
         return self._state
 
-    def _send_code(self, code, protocol, pulselength):
-        """Send the code with a specified pulselength."""
-        _LOGGER.info("Sending code: %s", code)
-        res = self._rfdevice.tx_code(code, protocol, pulselength)
-        if not res:
-            _LOGGER.error("Sending code %s failed", code)
-        return res
+    def _send_code(self, code_list, protocol, pulselength):
+        """Send the code(s) with a specified pulselength."""
+        _LOGGER.info("Sending code(s): %s", code_list)
+        for code in code_list:
+            self._rfdevice.tx_code(code, protocol, pulselength)
```

Comment:
Nevermind! To clarify, with performance reasons I meant that for some reason, when transmitting the specified codes via the rpi_rf.py script directly without going through HA, the codes would get transmitted fast enough so the switch will be triggered. At first, when I modified the HA component to be able to have support for multiple codes while preserving the return value, the codes all get executed properly, but the switch would not turn on or off.
I noticed that when I removed "bulk" to make it run faster that it would turn the switch on and off. This makes me think that the additional "bulk" of having to check the return value each time takes up so much processing power while running HA at the same time that it cannot send the codes fast enough one after each other. I think that maybe for the switch to recognize all of the codes as "one sequence" they cannot have a delay between them that is longer than a specific amount of time.